cloud-controller-manager/do/resources.go

```diff
@@ -229,3 +218,61 @@ func (r *ResourcesController) tagResources(res []godo.Resource) error {
 
 	return err
 }
+
+func (r *resources) retrieveLoadBalancer(ctx context.Context, service *corev1.Service) (*godo.LoadBalancer, error) {
```

Comment:
Is there a particular reason you defined/moved the methods on the `resources` struct? AFAICS, the receiver `r` is not used other than for calling methods.

`resources` is focused on updating tags on LB objects. I believe it'd make more sense to keep the functionality related to looking up LBs on the `loadBalancers` struct instead. It could also decrease the PR change set.

Reply:
@timoreimann the problem is that these function require access to data we hold in both, `resouces` and `loadbalancers`, but `resources` do not have access to `loadbalancers`, `loadbalancers` have access to `resources`. From what I understand `resources` represents DigtialOcean resources, and semantically it makes sense to me.

Moving this code to `loadbalancers` would result in code duplication, which might be acceptable, but my impression is that keeping this in one place makes it easier to understand the code. WDYT?

Reply:
By the required data that we hold in `reosuces` and `loadbalancers`. I mean the DO API client which is needed to retrieve a single load balancer resource / the list of load balancers.

Keeping this code in `resources` makes it possible to avoid a significant amount of duplication - otherwise we would need to duplicate code for retrieving load balancers between `resources` and `loadbalancers`.

One solution to this problem is to extracting DO API resources related to load balancers to a separate file, but I thought that `resources` is a good place where we could keep this code implemented, because my understanding is that indeed represents DO API resources :thinking:

Reply:
The name _resources_ is somewhat misleading: right now, the code in that file is about ensuring that tags are set on LBs. With that context in mind, I'm uncertain if it makes sense to have half of the LB-related code in `loadbalancers.go` and the other half in `resources.go`.

It wasn't immediately clear to me where the duplication would happen, so I took a stab at moving the LB functionality that's now in `resources.go` into `loadbalancers.go`. Turns out this was fairly easy after realizing that a few helper methods added do not really need a `*loadBalancers` receiver (i.e., they can be functions). Could you please take a look at my branch at https://github.com/timoreimann/digitalocean-cloud-controller-manager/tree/feature/gb/add-load-balancer-name-annotation and the commits I added? One is about moving the logic, and two more commits are about minor improvements I noticed might be worth doing.

If you agree that it makes sense to structure things like that, could I ask you to cherry-pick my commits into your PR branch? Afterwards, we can finish the code review.

By the way, I really like the refactoring you did around putting existing logic into functions. It makes the code much more readable. 💙

Reply:
Ahh, I somehow missed that down the road I decoupled `findLoadBalancerByName` from `gclient`. It indeed looks much better now @timoreimann. I will cherry pick commits :+1: Thanks for showing me that I was wrong :bowing_man: